src/plugins/intel_gpu/src/graph/concatenation.cpp

```diff
@@ -17,22 +17,22 @@ primitive_type_id concatenation::type_id() {
     return &instance;
 }
 
-layout concatenation_inst::calc_output_layout(concatenation_node const& node) {
-    auto desc = node.get_primitive();
+layout concatenation_inst::calc_output_layout(concatenation_node const& node, kernel_impl_params const& impl_param) {
+    auto desc = impl_param.typed_desc<concatenation>();
 
-    auto input_layout = node.input(0).get_output_layout();
+    auto input_layout = impl_param.input_layouts[0];
```

Comment:
Maybe we need a check here on input_layouts size. Otherwise, we may face segmentation fault in case of any bugs within these methods

Reply:
I merge this PR. Please add inputs count check in the new PR

Reply:
Okay, I'll create a separate PR to follow up this.